sql/core/src/main/scala/org/apache/spark/sql/execution/SubqueryBroadcastExec.scala

```diff
@@ -93,7 +94,9 @@ case class SubqueryBroadcastExec(
         val rows = if (broadcastRelation.keyIsUnique) {
           keyIter.toArray[InternalRow]
         } else {
-          keyIter.toArray[InternalRow].distinct
+          val set = new HashSet[InternalRow]()
+          keyIter.foreach(set.add(_))
+          set.toArray[InternalRow]
```

Comment:
Can we just:
```scala
keyIter.toStream.distinct.toArray[InternalRow]
```
?

Reply:
Thanks, modified

Reply:
`keyIter.toSet[InternalRow].toArray`?

Reply:
> `keyIter.toSet[InternalRow].toArray`?

It really should do this, otherwise it won't compile